vtr_flow/scripts/python_libs/vtr/parse_vtr_task.py

```diff
@@ -0,0 +1,602 @@
+    )
+
+    parser.add_argument("-run", default=None, type=str, help="")
+
+    parser.add_argument("-revision", default="", help="Revision number")
+
+    return parser
+
+
+def vtr_command_main(arg_list, prog=None):
+    """
+    Main function for parse_vtr_task
+    Parses in the results from run_vtr_task.py
+    """
+    # Load the arguments
+    args = vtr_command_argparser(prog).parse_args(arg_list)
+    try:
+        task_names = args.task
+
+        for list_file in args.list_file:
+            task_names += load_list_file(list_file)
+
+        config_files = [find_task_config_file(task_name) for task_name in task_names]
+
+        configs = [load_task_config(config_file) for config_file in config_files]
+        num_failed = 0
+
+        jobs = create_jobs(args, configs, after_run=True)
+        parse_tasks(configs, jobs)
+
+        if args.create_golden:
+            create_golden_results_for_tasks(configs)
+
+        if args.check_golden:
+            num_failed += check_golden_results_for_tasks(configs)
+
+        if args.calc_geomean:
+            summarize_qor(configs)
+            calc_geomean(args, configs)
+
+    except CommandError as error:
+        print("Error: {msg}".format(msg=error.msg))
+        print("\tfull command: ", error.cmd)
+        print("\treturncode  : ", error.returncode)
+        print("\tlog file    : ", error.log)
+        num_failed += 1
+    except InspectError as error:
+        print("Error: {msg}".format(msg=error.msg))
+        if error.filename:
+            print("\tfile: ", error.filename)
+        num_failed += 1
+    except VtrError as error:
+        print("Error:", error.msg)
+        num_failed += 1
+
+    return num_failed
+
+
+def parse_tasks(configs, jobs):
+    """
+    Parse the selection of tasks specified in configs and associated jobs
+    """
+    for config in configs:
+        config_jobs = [job for job in jobs if job.task_name() == config.task_name]
+        parse_task(config, config_jobs)
+
+
+def parse_task(config, config_jobs, flow_metrics_basename="parse_results.txt"):
+    """
+    Parse a single task run.
+
+    This generates a file parse_results.txt in the task's working directory,
+    which is an amalgam of the parse_rests.txt's produced by each job (flow invocation)
+    """
+    run_dir = find_latest_run_dir(config)
+
+    # Record max widths for pretty printing
+    max_arch_len = len("architecture")
+    max_circuit_len = len("circuit")
+    for job in config_jobs:
+        work_dir = job.work_dir(get_latest_run_dir(find_task_dir(config)))
+        if job.parse_command():
+            parse_filepath = str(PurePath(work_dir) / flow_metrics_basename)
+            with open(parse_filepath, "w+") as parse_file:
+                with redirect_stdout(parse_file):
+                    parse_vtr_flow(job.parse_command())
+        if job.second_parse_command():
+            parse_filepath = str(PurePath(work_dir) / "parse_results_2.txt")
```

Comment:
Please move arbitrary file names to top of file.

Reply:
This has been fixed